pages/beckman_cherrypick.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <b-container>
+    <h1>Beckman Cherrypick</h1>
+    <Alert ref="alert"></Alert>
+    <b-card no-body>
+      <b-tabs card>
+        <b-tab title="Create">
+          <b-card
+            title="Create Destination Plate"
+            sub-title="Generate destination plate from DART data so it can continue in pipeline partially filled."
+          >
+            <BeckmanCherrypickForm
+              v-slot="{ form, formInvalid }"
+              :action="'create'"
+              :robots="robots"
+            >
+              <b-button
+                variant="success"
+                :disabled="formInvalid"
+                @click="create(form)"
+                >Create Destination Plate</b-button
+              >
+            </BeckmanCherrypickForm>
+          </b-card>
+        </b-tab>
+
+        <b-tab title="Fail">
+          <b-card
+            title="Fail Destination Plate"
+            sub-title="Fail destination plate with a reason."
+          >
+            <BeckmanCherrypickForm
+              v-slot="{ form, formInvalid }"
+              :action="'fail'"
+              :robots="robots"
+              :failure-types="failureTypes"
+            >
+              <b-button
+                variant="danger"
+                :disabled="formInvalid"
+                @click="fail(form)"
+                >Fail Destination Plate</b-button
+              >
+            </BeckmanCherrypickForm>
+          </b-card>
+        </b-tab>
+      </b-tabs>
+    </b-card>
+  </b-container>
+</template>
+
+<script>
+// https://ssg-confluence.internal.sanger.ac.uk/display/PSDPUB/Cherrypicking+Events
+
+import BeckmanCherrypickForm from '@/components/BeckmanCherrypickForm'
+import lighthouse from '@/modules/lighthouse_service'
+import Alert from '@/components/Alert'
+
+export default {
+  components: {
+    BeckmanCherrypickForm,
+    Alert
+  },
+  data() {
+    return {
+      robots: [],
+      failureTypes: []
+    }
+  },
+  async mounted() {
+    await this.getRobots()
+    await this.getFailureTypes()
+  },
+  methods: {
```

Comment:
Advisory: the getRobots() and getFailureTypes() seem essentially the same method apart from the method. You could have a single function for this

Reply:
Yes, will do

Reply:
Excellent. Exactly the way I was thinking how to do it. Proper JavaScript. Passing the function and using dynamic object keys.